### PR TITLE
Don't rely on EnsureOOBFramework

### PR DESCRIPTION
**Pull request by ericstj**
Explicitly place assets in netcore50 folder where needed.

There are a few libraries which will stop using the live build after this change.
System.Diagnostics.Tools - single attribute
System.Reflection.Context - not supported assembly
System.Runtime.InteropServices.WindowsRuntime - facade
System.Runtime.Serialization.Json - facade
System.Runtime.Serialization.Xml - facade
System.Security.Principal - simple types

Let me know if you think I should re-enable the live build for any of these.
@weshaggard 

BuildTools PR that this enables: https://github.com/dotnet/buildtools/pull/1054

### CODE REVIEW

File: src/Microsoft.VisualBasic/ref/Microsoft.VisualBasic.csproj [hunk 1]
```diff
@@ -5,6 +5,11 @@
     <AssemblyVersion>10.0.0.0</AssemblyVersion>
     <OutputType>Library</OutputType>
     <NuGetTargetMoniker>.NETStandard,Version=v1.1</NuGetTargetMoniker>
+    <!-- Xamarin frameworks claim to support PCL profiles which include this assembly, 
+         but do not include this assembly in the targeting pack.
+         This causes NuGet to give them the placeholder from the portable-* folder.
+         Instead give them this assembly.  -->
+    <PackageTargetFramework>netstandard1.1;MonoAndroid10;MonoTouch10;xamarinios10;xamarintvos10;xamarinwatchos10</PackageTargetFramework>
```
Comment (weshaggard) [inline src/Microsoft.VisualBasic/ref/Microsoft.VisualBasic.csproj]: Do we need a property that contains the list of Xamarin targets?

File: src/Microsoft.CSharp/src/Microsoft.CSharp.csproj [hunk 2]
```diff
@@ -8,6 +8,7 @@
     <!-- dotnet/corefx#3128 tracks removing this exclusion -->
     <ExcludeLocalizationImport Condition="'$(TargetsWindows)' != 'true'">true</ExcludeLocalizationImport>
     <NuGetTargetMoniker Condition="'$(TargetGroup)' == ''">.NETStandard,Version=v1.3</NuGetTargetMoniker>
+    <PackageTargetFramework Condition="'$(TargetGroup)' == ''">netstandard1.3;netcore50</PackageTargetFramework>
```
Comment (weshaggard) [inline src/Microsoft.CSharp/src/Microsoft.CSharp.csproj]: Will this conflict with the Ensure OOB task or will it no-op?

Comment (ericstj) [inline src/Microsoft.CSharp/src/Microsoft.CSharp.csproj]: Won't conflict.  Explicit assets always win.  That's why this commit works even without the buildtools change.

File: src/Microsoft.VisualBasic/ref/Microsoft.VisualBasic.csproj (hunk 1 above)
Comment (ericstj) [inline src/Microsoft.VisualBasic/ref/Microsoft.VisualBasic.csproj]: We do have one, but its only in the packaging targets which are only imported in pkgproj's.  I'd prefer not to duplicate that.  Even further I'd prefer if these assemblies were just copied into Xamarin's frameworks so we don't have to special case like this.  Both this and System.Reflection.Context are part of PCL profiles which they support but the assemblies are missing from their targeting pack and framework.

**Pull request merged**